performance-harness/proposals/performance_testing_harness_stage_3.md

```diff
@@ -0,0 +1,107 @@
+# Performance Testing Harness Stage 3 Planning
+
+Roadmap for the development of a **Performance Testing Harness** -- Stage 3.
+
+Requirements are being curated for the development of a performance testing and evaluation harness here: [Performance Testing Harness Requirements & Planning & Staging](https://github.com/eosnetworkfoundation/product/tree/main/performance-harness/proposals)
+
+
+## Stage 3
+
+###   Multi-producer & "relay" or "canary" node configuration support (Carried over from Stage 2)

```

Comment:
I think we should consider breaking this in to multiple stages.  The idea of a full simulated network has less immediate value than the ability to baseline and track the "normal" modes of operation.  

My understanding is that we currently have a single node that is taking transactions in over P2P and emitting blocks. That seems like a good "block producer" operation mode.  

I think this section could be scoped down to a "relay node" which primarily takes transactions and blocks as inputs over the P2P protocol and relays them to peers. I would like to see if we can benchmark that mode in isolation (mocking the sending and receiving peers as much as possible).  While the mock receivers/transmitters can be written in anything including c++ using leap code, I will also point out that there are [golang](https://github.com/eoscanada/eos-go/blob/develop/p2p.go) and [typescript](https://github.com/greymass/eosio-core/pull/51) clients for P2P as well (maybe more) 

Once we get more "modes" rigged up then we would investigate a more real-world simulation of a full network.